packages/design-picker/src/utils/designs.ts

```diff
@@ -11,14 +11,17 @@ export const getDesignPreviewUrl = (
 	if ( [ 'hannah', 'riley', 'gilbert' ].indexOf( slug ) >= 0 ) {
 		return `https://${ slug }starter.wordpress.com`;
 	}
-
 	let url = addQueryArgs( 'https://public-api.wordpress.com/wpcom/v2/block-previews/site', {
 		stylesheet: recipe?.stylesheet,
 		pattern_ids: recipe?.patternIds?.join( ',' ),
 		language: options.language,
 		viewport_height: 700,
 	} );
 
+	if ( recipe?.patternIds?.length ) {
+		url += '&source_site=patternboilerplates.wordpress.com';
```

Comment:
Ideally, we should add this to the args object above, rather than appending the query param manually in the final string, so that it can be encoded properly.

(We append the query param directly for `site_title` as a special case -- because of the issue mentioned in the comments above it)

Can we change it to something like

```
{
		stylesheet: recipe?.stylesheet,
		pattern_ids: recipe?.patternIds?.join( ',' ),
        source_site: recipe?.patternIds?.length ? 'patternboilerplates.wordpress.com' : undefined,
		language: options.language,
		viewport_height: 700,
	}
```

or something similar?

Reply:
But actually I don't see any harm in always passing the source site? This will simplify the logic. 😄

Reply:
Yep, adding args by `addQueryArgs` is better except some special cases like `site_title`

Reply:
I was thinking not to add it because then the caching would change for the other previews. But actually this doesn't matter since it would only change once 😅 I'll revert to the simpler code.